mdm/acknowledge.go

```diff
@@ -6,12 +6,11 @@ import (
 	"time"
 
 	"github.com/go-kit/kit/endpoint"
-	"github.com/groob/plist"
 	"github.com/pkg/errors"
 	uuid "github.com/satori/go.uuid"
 )
 
-func (svc *MDMService) Acknowledge(ctx context.Context, req AcknowledgeEvent) (payload []byte, err error) {
+func (svc *MDMService) Acknowledge(ctx context.Context, req AcknowledgeEvent, _ Device) (payload []byte, err error) {
```

Comment:
I would prefer to see the Certificate either as fields on `AcknowledgeEvent`/`CheckinEvent` or as a context value. I don't see a reason we should add the new param. It's only used in middleware, and would be ignored by the actual implementation, so it would be cleaner not to force it in the function signature.

Reply:
I chose to put it on the context. I don't feel good about putting elements on the events without also having them marshalled. Just seems like a hidden/undiscoverable thing.